dnf5-plugins/config-manager_plugin/unsetopt.cpp

```diff
@@ -107,48 +114,88 @@ void ConfigManagerUnsetOptCommand::configure() {
     const auto & config = ctx.base.get_config();
 
     // Remove options from main configuration file.
-    const auto & cfg_filepath = get_config_file_path(ctx.base.get_config());
-    if (!main_opts_to_remove.empty() && std::filesystem::exists(cfg_filepath)) {
-        ConfigParser parser;
-        bool changed = false;
-
-        parser.read(cfg_filepath);
-
-        changed |= remove_from_config(parser, "main", main_opts_to_remove);
+    if (!main_opts_to_remove.empty()) {
+        const auto & cfg_filepath = get_config_file_path(ctx.base.get_config());
+        if (std::filesystem::exists(cfg_filepath)) {
+            ConfigParser parser;
+            bool changed = false;
+            std::set<std::string> used_keys;
+
+            parser.read(cfg_filepath);
+
+            changed |= remove_from_config(parser, "main", main_opts_to_remove, used_keys);
+
+            // Generate warning for unused options
+            for (const auto & key : main_opts_to_remove) {
+                if (!used_keys.contains(key)) {
+                    ctx.base.get_logger()->warning(
+                        "config-manager: Request to remove not present main option from config file: {}", key);
+                }
+            }
 
-        if (changed) {
-            parser.write(cfg_filepath, false);
+            if (changed) {
+                parser.write(cfg_filepath, false);
+            }
+        } else {
+            ctx.base.get_logger()->warning(
+                "config-manager: Request to remove main option but config file not found: {}", cfg_filepath.string());
         }
     }
 
-    auto repos_override_file_path = get_config_manager_repos_override_file_path(config);
-
     // Remove options from repositories overrides configuration file, remove empty sections.
-    if (!in_repos_opts_to_remove.empty() && std::filesystem::exists(repos_override_file_path)) {
-        ConfigParser parser;
-        bool changed = false;
-        parser.read(repos_override_file_path);
+    if (!in_repos_opts_to_remove.empty()) {
+        const auto & repos_override_file_path = get_config_manager_repos_override_file_path(config);
```

Comment:
This is not a problem of this PR, but I encountered this:

When I run `dnf5 config-manager unsetopt fedora.gpgcheck` (or any other option), there is no output, but nothing gets changed. Only when I finally looked at logs, I noticed there is `config-manager: Request to remove repository option but config file not found: /etc/dnf/repos.override.d/99-config_manager.repo`.

Shouldn't this file get created if it doesn't exist yet? Or am I missing something?

Moreover, this makes me think even more that the warnings should get also on the stderr, because it took me some time to figure out why nothing is happening.

Reply:
@pkratoch 
>Shouldn't this file get created if it doesn't exist yet?

No. The file contains overrides of repository configuration. A non-existent file has the same meaning as an empty file - no overrides are defined.

Reply:
Ah, right. It doesn't make sense to create the file for unsetting the options. I was confused only because I initially expected the original repo file to be modified and seemingly nothing happened when I run the `unsetopt` command. But now I see it only works over the overrides and the error message makes that clear.